File: packages/sdk-js/src/resources/reports.ts
import { BaseResource } from './base';
import { AxiosRequestConfig } from 'axios';

// Tenant-scoped reporting. Each endpoint aggregates real data for the caller's
// tenant into a fixed, named report — there is no list/get-by-id surface.
//
// Mounted at /api/reports on the Community Edition server.

export class ReportsResource extends BaseResource {
  protected readonly RESOURCE_PATH = 'reports';

  /** Unified activity feed. GET /api/reports/audit-logs */
  async getAuditLogs(params?: Record<string, any>, config?: AxiosRequestConfig): Promise<any> {
    return this.httpClient.get<any>(`${this.RESOURCE_PATH}/audit-logs`, { ...config, params });
  }

  /** User-activity report (stats, sessions, actions). GET /api/reports/user-activity */
  async getUserActivity(params?: Record<string, any>, config?: AxiosRequestConfig): Promise<any> {
    return this.httpClient.get<any>(`${this.RESOURCE_PATH}/user-activity`, { ...config, params });
  }

  /** Resource-usage report (real inventory). GET /api/reports/resource-usage */
  async getResourceUsage(params?: Record<string, any>, config?: AxiosRequestConfig): Promise<any> {
    return this.httpClient.get<any>(`${this.RESOURCE_PATH}/resource-usage`, { ...config, params });
  }

  /** Security-overview report (derived posture). GET /api/reports/security-overview */
  async getSecurityOverview(params?: Record<string, any>, config?: AxiosRequestConfig): Promise<any> {
    return this.httpClient.get<any>(`${this.RESOURCE_PATH}/security-overview`, { ...config, params });
  }

  /** Compliance report (frameworks + controls). GET /api/reports/compliance */
  async getCompliance(params?: Record<string, any>, config?: AxiosRequestConfig): Promise<any> {
    return this.httpClient.get<any>(`${this.RESOURCE_PATH}/compliance`, { ...config, params });
  }
}

File: packages/sdk-python/veltrix_sdk/resources/reports.py
# -*- coding: utf-8 -*-

from .base import BaseResource


class ReportsResource(BaseResource):
    """Handles tenant-scoped reporting (``/api/reports``).

    Each endpoint aggregates real data for the caller's tenant into a fixed,
    named report -- there is no list/get-by-id surface.
    """
    RESOURCE_PATH = "reports"

    def get_audit_logs(self, params=None, **kwargs):
        """Unified activity feed (GET /audit-logs)."""
        return self._http_client.get(f"{self.RESOURCE_PATH}/audit-logs", params=params, **kwargs)

    def get_user_activity(self, params=None, **kwargs):
        """User-activity report -- stats, sessions, actions (GET /user-activity)."""
        return self._http_client.get(f"{self.RESOURCE_PATH}/user-activity", params=params, **kwargs)

    def get_resource_usage(self, params=None, **kwargs):
        """Resource-usage report -- real inventory (GET /resource-usage)."""
        return self._http_client.get(f"{self.RESOURCE_PATH}/resource-usage", params=params, **kwargs)

    def get_security_overview(self, params=None, **kwargs):
        """Security-overview report -- derived posture (GET /security-overview)."""
        return self._http_client.get(f"{self.RESOURCE_PATH}/security-overview", params=params, **kwargs)

    def get_compliance(self, params=None, **kwargs):
        """Compliance report -- frameworks + controls (GET /compliance)."""
        return self._http_client.get(f"{self.RESOURCE_PATH}/compliance", params=params, **kwargs)
